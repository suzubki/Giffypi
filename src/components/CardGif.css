.card-margin{
    padding: 2rem;
}

.card_image--margin{
    margin: 2px auto 0 auto;
}