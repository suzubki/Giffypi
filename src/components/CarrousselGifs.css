.card-margin {
    padding: 2rem;
}

.card_image--margin {
    margin: 0 auto 0 auto;
}
