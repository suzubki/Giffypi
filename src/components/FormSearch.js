import React from "react";
import { Input, Button, Form } from "reactstrap";
import { useSearchParams } from "react-router-dom";
// import { useGifsSearch } from "../helpers/useGifsSearch";

export const FormSearch = () => {
    //  useSearchParams
    let [searchParams, setSearchParams] = useSearchParams();
    let giphy = searchParams.get("giphy");

    const handleSubmit = (e) => {
        e.preventDefault();

        let formData = new FormData(e.currentTarget);
        let newGiphy = formData.get("giphy");

        if (!newGiphy) return;
        if (giphy === newGiphy) return;

        setSearchParams({ giphy: newGiphy });
    };

    return (
        <div>
            <Form
                onSubmit={handleSubmit}
                className="p-2"
                style={{
                    maxWidth: "1000px",
                    marginLeft: "auto",
                    marginRight: "auto",
                }}
            >
                <Input
                    defaultValue={giphy || ""}
                    name="giphy"
                    autoComplete="off"
                />
                <Button type="submit" className="mt-2" color="primary">
                    Search
                </Button>
            </Form>
        </div>
    );
};
