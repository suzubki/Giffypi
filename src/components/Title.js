import React from "react";
import "bootstrap";

export const Title = React.memo(() => {
    return (
        <div>
            <h1 className="p-4 text-warning text-center">Giffapi</h1>
        </div>
    );
});
