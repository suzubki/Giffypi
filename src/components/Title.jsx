import React from "react";
import "bootstrap";

export const Title = () => {
  return (
    <div>
      <h1 className="p-4 text-white">Giffapi</h1>
    </div>
  );
};
