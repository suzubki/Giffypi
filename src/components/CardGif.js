import React from "react";
import PropTypes from 'prop-types'

// React slick (carousel)
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

// Reactstrap
import { Card } from "reactstrap";

// CSS
import "./CardGif.css";

export const CardGif = ({ objectData }) => {

  let { data, loading } = objectData

  // Slick settings
  const settings = {
    infinite: true,
    autoplay: true,
    speed: 2000,
    autoplaySpeed: 2000,
    slidesToShow: 4,
    slidesToScroll: 1,
    centerMode: true,
    centerPadding: "10px",
    initialSlide: 0,
    pauseOnHover: true,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
          infinite: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };

  // Render card gif
  return (
    <>
      <Slider {...settings}>
        {
        loading && data.map((gif) => (
            <Card key={gif.id}>
              <img
                alt="Card cap"
                className="card_image--margin"
                src={gif.images.original.url}
                width="90%"
              />
            </Card>
        ))}
      </Slider>
    </>
  );
};


CardGif.propTypes = {
  objectData: PropTypes.object.isRequired
}

CardGif.defaultProps = {
  objectData: {data: [], loading: Boolean}
}

export default CardGif;
