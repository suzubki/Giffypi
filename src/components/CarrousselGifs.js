import React, { useContext, useEffect } from "react";

// React slick (carousel)
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

// Reactstrap
import { Card } from "reactstrap";

// CSS
import "./CarrousselGifs.css";
import getGifsTrending from "../helpers/getGifsTrending";
import { DataContext } from "../helpers/dataContext";
import { useSearchParams } from "react-router-dom";
import { getGifsSearch } from "../helpers/getGifsSearch";

export const CarrousselGifs = () => {
    const { dataGifs, setDataGifs } = useContext(DataContext);
    const [searchParams] = useSearchParams();

    const query = searchParams.get("giphy");

    // Call API - Trending and Search endpoint and monitorize the call api
    useEffect(() => {
        if (!query) {
            console.log("asdasds");
            getGifsTrending()
                .then((res) => {
                    console.log("Llamado a gifsTrending");
                    setDataGifs({
                        data: [...res],
                        loading: false,
                    });
                })
                .catch((err) => {
                    console.log(err);
                });
        }

        const controller = new AbortController();

        if (query) {
            setDataGifs({ data: [], loading: true });
            getGifsSearch(query, controller)
                .then((res) => {
                    setDataGifs({
                        data: [...res],
                        loading: false,
                    });
                })
                .catch((err) => {
                    console.error(err);
                });
        }

        return () => {
            controller.abort();
        };
    }, [setDataGifs, query]);

    // Slick settings
    const settings = {
        infinite: true,
        autoplay: true,
        adaptiveHeigh: true,
        speed: 2000,
        autoplaySpeed: 2000,
        slidesToShow: 4,
        slidesToScroll: 1,
        centerMode: true,
        centerPadding: "10px",
        initialSlide: 0,
        pauseOnHover: true,
        responsive: [
            {
                breakpoint: 1024,
                settings: {
                    slidesToShow: 3,
                    slidesToScroll: 1,
                    infinite: true,
                },
            },
            {
                breakpoint: 600,
                settings: {
                    slidesToShow: 2,
                    slidesToScroll: 1,
                    initialSlide: 2,
                },
            },
            {
                breakpoint: 480,
                settings: {
                    slidesToShow: 1,
                    slidesToScroll: 1,
                },
            },
        ],
    };

    // Render card gif
    return (
        <div
            className="text-center"
            style={{
                marginLeft: "auto",
                marginRight: "auto",
                maxWidth: "80vw",
                height: "auto",
            }}
        >
            {/* data found = 0 and loading = false ? No elements to render */}
            {dataGifs.data?.length === 0 && dataGifs.loading === false && (
                <h1 className="text-white text-center mt-4 ">
                    No se encontraron resultados
                </h1>
            )}

            {/* Show 'Cargando...' or render Slick */}
            {dataGifs.loading ? (
                <h1 className="text-white text-center mt-4 ">Cargando...</h1>
            ) : (
                <Slider {...settings}>
                    {dataGifs.data?.map((gif) => (
                        <Card key={gif.id}>
                            <img
                                alt="Card cap"
                                className="card_image--margin"
                                src={gif.images.original.url}
                                width="90%"
                            />
                        </Card>
                    ))}
                </Slider>
            )}
        </div>
    );
};

export default CarrousselGifs;
