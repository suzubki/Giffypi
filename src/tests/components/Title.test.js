import React from 'react';
import {shallow} from 'enzyme';
import {Title} from '../../components/Title';


describe('DeberÃ­a mostrar solo el wrapper', () => { 
    test('mostrando el wrapper', () => { 
        const wrapper = shallow( <Title /> )

        expect(wrapper).toMatchSnapshot()

     })
 })