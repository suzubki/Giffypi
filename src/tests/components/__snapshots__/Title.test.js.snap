// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deber√≠a mostrar solo el wrapper mostrando el wrapper 1`] = `
<div>
  <h1
    className="p-4 text-white"
  >
    Giffapi
  </h1>
</div>
`;
